docker/src/main/java/brooklyn/entity/container/docker/DockerInfrastructure.java

```diff
@@ -86,12 +86,15 @@
     @SetFromFlag("enableSdn")
     ConfigKey<Boolean> SDN_ENABLE = SdnAttributes.SDN_ENABLE;
 
+    @SuppressWarnings("serial")
     @SetFromFlag("sdnProviderSpec")
-    ConfigKey<EntitySpec> SDN_PROVIDER_SPEC = ConfigKeys.newConfigKey(EntitySpec.class, "sdn.provider.spec", "SDN provider entity specification");
+    ConfigKey<EntitySpec<?>> SDN_PROVIDER_SPEC = ConfigKeys.newConfigKey(
+            new TypeToken<EntitySpec<?>>() {}, "sdn.provider.spec", "SDN provider entity specification");
 
+    @SuppressWarnings("serial")
     @SetFromFlag("hostSpec")
-    AttributeSensorAndConfigKey<EntitySpec, EntitySpec> DOCKER_HOST_SPEC = ConfigKeys.newSensorAndConfigKey(
-            EntitySpec.class, "docker.host.spec", "Specification to use when creating child Docker Hosts",
+    AttributeSensorAndConfigKey<EntitySpec<?>, EntitySpec<?>> DOCKER_HOST_SPEC = ConfigKeys.newSensorAndConfigKey(
+            new TypeToken<EntitySpec<?>>() {}, "docker.host.spec", "Specification to use when creating child Docker Hosts",
```

Comment:
@grkvlt I've stripped out the generics that I'd introduced from the interfaces, but have not figured out how to avoid the cast to `EntitySpec<?>` in [DockerInfrastructureImpl.java#L185](https://github.com/johnmccabe/clocker/blob/0.9.0-update/docker/src/main/java/brooklyn/entity/container/docker/DockerInfrastructureImpl.java#L185]) and [MesosClusterImpl.java#L149](https://github.com/johnmccabe/clocker/blob/0.9.0-update/mesos/src/main/java/brooklyn/entity/mesos/MesosClusterImpl.java#L149).